### PR TITLE
Various documentation improvements

### PR DESCRIPTION
**Pull request by airwoodix**
### Summary

Improve on the documentation:

- sidebar: add link to the API reference
- quick-start: add minimum installation instructions
- guide: add separate section for the circuit transformations performed behind the API wall
- guide: add note on variable-angle `RXX` gates.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -11,6 +11,12 @@ hardware.
 Quick start
 -----------
 
+Install the latest release from the `PyPI <https://pypi.org/project/qiskit-aqt-provider>`_:
+
+.. code-block:: bash
+
+  pip install qiskit-aqt-provider
+
```
[the remark this answers is not in the document]
Comment (airwoodix) [reply docs/index.rst:19]: I'm a bit relunctant to duplicate information from `pyproject.toml` here, as the docs would easily drift.
If the virtualenv is not empty and `pip` is invoked without `--upgrade`, it should either realize that the available version is ok, or error on a dependency conflict.

[the remark this answers is not in the document]
Comment (fg-aqt) [reply docs/index.rst:19]: Alright, makes sense. What do you think about a "Note" that says something like "There can be dependencies pinned to specific versions, make sure there are no conflicting package versions in your environment"

[the remark this answers is not in the document]
Comment (airwoodix) [reply docs/index.rst:19]: How about baf4e89?

[the remark this answers is not in the document]
Comment (fg-aqt) [reply docs/index.rst:19]: Looks good :+1:

**Review by fg-aqt — APPROVED** [review]

**Pull request merged**